src/Toml/Printer.hs

```diff
@@ -20,11 +22,28 @@ import qualified Data.HashMap.Strict as HashMap
 import qualified Data.List.NonEmpty as NonEmpty
 import qualified Data.Text as Text
 
--- Tab is equal to 2 spaces now.
-tab :: Int -> Text
-tab n = Text.cons '\n' (Text.replicate (2*n) " ")
+{- | Configures the pretty printer. -}
+data PrintOptions = PrintOptions
+    { shouldSort :: Bool  -- ^ should table keys be sorted ot shouldn't
+    , indent     :: Int   -- ^ indentation size
+    } deriving (Show)
 
-{- | Converts 'TOML' type into 'Text'.
+{- | Default printing options. -}
+defaultOptions :: PrintOptions
+defaultOptions = PrintOptions True 2
+
+-- Returns an indentation prefix
+tabWith :: PrintOptions -> Int -> Text
+tabWith options n =
+    Text.cons '\n' (Text.replicate (n * indent options) " ")
```

Comment:
`n` argument is redundant here and there's no need to multiply by `n`. In the code you assign `nextI` variable (which is responsible for indentation) to the value `i + indent options` and you pass `options` and `i` to the `tab` function. So, for example, if default indentation is `4` then layers will have the following indents:

* Nested 0: `indent = 4`, `n = 0` => tab is 0 spaces
* Nested 1: `indent = 4`, `n = 4` => tab is 16 spaces
* Nested 2: `indent = 4`, `n = 8` => tab is 32 spaces
* and so on

Reply:
Oh, I thought about it, but forgot to drop `+ indent options`...

Reply:
Now `n` means only a level of indentation (as it was before)